public/migrate.sh

```diff
@@ -20,85 +23,96 @@ os_id=$(grep -E '^ID=' /etc/os-release | sed -e 's/ID=//g')
 
-echo "Installing Terra..."
+echo " ...[2/3] Terra"
 trace sudo dnf install -y --repofrompath 'terra,https://repos.fyralabs.com/terra$releasever' --setopt='terra.gpgkey=https://repos.fyralabs.com/terra$releasever/key.asc' terra-release
 
-echo "Installing the Ultramarine repositories..."
+echo " ...[3/3] Ultramarine Repositories"
 trace sudo dnf --nogpgcheck --repofrompath "ultramarine,https://repos.fyralabs.com/um${os_version}/" install -y ultramarine-repos-common ultramarine-repos
 
-echo "Repo installation complete."
-echo
 
-echo "Converting to Ultramarine..."
+echo "■ (3/3) Converting to Ultramarine..."
 if [[ ${os_variant} = "workstation" ]]; then
+  echo ' ... Detected Fedora Workstation'
   trace sudo dnf swap -y fedora-release-common ultramarine-release-gnome --allowerasing
   trace sudo dnf group install --allowerasing -y ultramarine-gnome-product-environment
   trace sudo dnf group remove -y workstation-product-environment
 elif [[ ${os_variant} = "kde" ]]; then
+  echo ' ... Detected Fedora KDE Spin'
   trace sudo dnf swap -y fedora-release-common ultramarine-release-kde --allowerasing
   trace sudo dnf group install --allowerasing -y ultramarine-kde-product-environment
   trace sudo dnf group remove -y kde-desktop-environment
 elif [[ ${os_variant} = "budgie" ]]; then
+  echo ' ... Detected Fedora Budgie Spin'
   trace sudo dnf swap -y fedora-release-common ultramarine-release-flagship --allowerasing
   trace sudo dnf group install --allowerasing -y ultramarine-flagship-product-environment
   trace sudo dnf group remove -y budgie-desktop-environment
+elif [[ ${os_variant} = "xfce" ]]; then
+  echo ' ... Detected Fedora XFCE Spin'
+  trace sudo dnf swap -y fedora-release-common ultramarine-release-xfce --allowerasing
+  trace sudo dnf group install -y ultramarine-xfce-product-environment
+  trace sudo dnf group remove -y xfce-desktop-environment
 else # If the variant is unknown or doesn't have an equivalent in Ultramarine
+  echo ' ... Falling back to `ultramarine-release-common`'
   trace sudo dnf swap -y fedora-release-common ultramarine-release-common --allowerasing
   trace sudo dnf group install --allowerasing -y ultramarine-product-common
 fi
 trace sudo dnf swap -y fedora-logos ultramarine-logos --allowerasing
 
-echo "Migration complete! Please reboot your system.
+echo '┏━━━           ──────  '
+echo '  Migration Complete!  '
+echo '  ──────           ━━━┛'
+echo "You may now reboot your system.
 The next Linux kernel update will make your system entry appear as Ultramarine Linux, but now you're already running Ultramarine Linux."
 
-read -p "Would you like to also generate a new initramfs? (generate an Ultramarine boot entry) (y/n) " -r REPLY
+read -n 1 -p "□ Would you like to also generate a new initramfs? (generate an Ultramarine boot entry) (y/N) " -r REPLY
 
 if [[ ${REPLY} =~ ^[Yy]$ ]]; then
-  echo "Generating new initramfs..."
+  echo "■ Generating new initramfs..."
```

Comment:
add a blank echo statement above, otherwise this prints on the same line as the prompt